fx-src/FirefoxStorage.js

```diff
@@ -163,12 +165,14 @@ export default class FirefoxAdapter extends BaseAdapter {
 
     return conn.executeTransaction(function* doExecuteTransaction() {
       // Preload specified records from DB, within transaction.
-      const parameters = {
-        collection_name: collection,
-        // XXX: would be easier if a list of strings could be bound instead.
-        record_ids: options.preload.join("','")
-      };
-      const rows = yield conn.execute(statements.listRecordsById, parameters);
+      const parameters = [
+        collection,
+        ...options.preload
+      ];
+      const placeholders = options.preload.map(_ => "?");
+      const stmt = statements.listRecordsById + "(" + placeholders.join(",") + ");";
```

Comment:
Right. The previous version wasn't working, and this one does, so that makes it better :) The old version tried to generate a string that would generate a list by "unescaping" the commas, but of course it actually got escaped because it was fed into the query using a placeholder, so only turned into one giant string rather than a list of strings.

Yes, this is horrible, but I'm not super worried about the potential for a SQL injection attack -- we aren't doing interpolation ourselves, we're just generating a string with a variable number of placeholders. Substituting the placeholders for actual values is still done by the SQL library.

Note that I think we use a similar technique in the Kinto server at https://github.com/Kinto/kinto/blob/master/kinto/core/permission/postgresql/__init__.py#L285-L324.